### PR TITLE
[Release] Add 2.9.2 release logs and fix release log script

### PR DESCRIPTION
**Pull request by GeneDer**
## Why are these changes needed?

Adds performance logs for [fce7a36180](https://github.com/ray-project/ray/commit/fce7a361807580953364e2da964f9498f3123bf9)

```
(ray) gene@geneanycale2023 release_logs % python compare_perf_metrics 2.9.1 2.9.2
REGRESSION 8.15%: client__put_gigabytes (THROUGHPUT) regresses from 0.12938700124608027 to 0.1188387933899692 (8.15%) in 2.9.2/microbenchmark.json
REGRESSION 6.75%: client__get_calls (THROUGHPUT) regresses from 1125.9811184876232 to 1050.0307192769721 (6.75%) in 2.9.2/microbenchmark.json
REGRESSION 5.05%: single_client_tasks_async (THROUGHPUT) regresses from 8592.305437413408 to 8158.711929873504 (5.05%) in 2.9.2/microbenchmark.json
REGRESSION 4.70%: multi_client_put_gigabytes (THROUGHPUT) regresses from 35.263877895454804 to 33.605927354752815 (4.70%) in 2.9.2/microbenchmark.json
REGRESSION 1.76%: client__tasks_and_get_batch (THROUGHPUT) regresses from 0.9605952574022111 to 0.9437038816702141 (1.76%) in 2.9.2/microbenchmark.json
REGRESSION 1.66%: tasks_per_second (THROUGHPUT) regresses from 326.73502105628876 to 321.31153016474485 (1.66%) in 2.9.2/benchmarks/many_nodes.json
REGRESSION 0.11%: n_n_actor_calls_with_arg_async (THROUGHPUT) regresses from 2861.4449722009826 to 2858.2486735378648 (0.11%) in 2.9.2/microbenchmark.json
REGRESSION 237.49%: dashboard_p95_latency_ms (LATENCY) regresses from 9.013 to 30.418 (237.49%) in 2.9.2/benchmarks/many_nodes.json
REGRESSION 76.92%: dashboard_p50_latency_ms (LATENCY) regresses from 25.07 to 44.354 (76.92%) in 2.9.2/benchmarks/many_actors.json
REGRESSION 39.79%: dashboard_p99_latency_ms (LATENCY) regresses from 10083.034 to 14095.515 (39.79%) in 2.9.2/benchmarks/many_tasks.json
REGRESSION 32.51%: dashboard_p95_latency_ms (LATENCY) regresses from 6287.46 to 8331.543 (32.51%) in 2.9.2/benchmarks/many_tasks.json
REGRESSION 4.98%: 3000_returns_time (LATENCY) regresses from 6.457862126000009 to 6.779333391999998 (4.98%) in 2.9.2/scalability/single_node.json
REGRESSION 4.00%: 10000_get_time (LATENCY) regresses from 24.095646799999997 to 25.059263131999984 (4.00%) in 2.9.2/scalability/single_node.json
REGRESSION 2.82%: avg_iteration_time (LATENCY) regresses from 1.5108911991119385 to 1.5535523080825806 (2.82%) in 2.9.2/stress_tests/stress_test_dead_actors.json
REGRESSION 2.43%: 1000000_queued_time (LATENCY) regresses from 188.693754931 to 193.28140517800003 (2.43%) in 2.9.2/scalability/single_node.json
REGRESSION 1.29%: stage_4_spread (LATENCY) regresses from 0.637963224514887 to 0.64618222013863 (1.29%) in 2.9.2/stress_tests/stress_test_many_tasks.json
REGRESSION 0.25%: dashboard_p50_latency_ms (LATENCY) regresses from 4.359 to 4.37 (0.25%) in 2.9.2/benchmarks/many_nodes.json
REGRESSION 0.23%: 107374182400_large_object_time (LATENCY) regresses from 29.431489800999998 to 29.50047878800001 (0.23%) in 2.9.2/scalability/single_node.json
REGRESSION 0.21%: avg_pg_remove_time_ms (LATENCY) regresses from 0.8617512267259182 to 0.863579004504134 (0.21%) in 2.9.2/stress_tests/stress_test_placement_group.json
```

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by GeneDer** [comment]
@jjyao @rynewang Can you help checking the regression please?

File: release/release_logs/fetch_release_logs.py [hunk 1]
```diff
@@ -38,7 +38,7 @@
 from pybuildkite.buildkite import Buildkite
 
 BUILDKITE_ORGANIZATION = "ray-project"
-BUILDKITE_PIPELINE = "release-tests-branch"
+BUILDKITE_PIPELINE = "release"
```
Comment (GeneDer) [inline release/release_logs/fetch_release_logs.py:41]: @can-anyscale For some reason the pipeline name has changed. The script only works if I update this name 😅 Let me know if we want to just update the script or if we should change make the pipeline name

**Comment by rynewang** [comment]
@GeneDer where are these numbers come from? This release does not have a lot of Core patches so we expect little change. Could be variance? Is there a way we rerun those regressions?

And btw, @cadedaniel mentioned` dashboard_p95_latency_ms` in particularly is very noisy, should ignore

**Comment by GeneDer** [comment]
@rynewang Those are typically from the release tests. https://buildkite.com/ray-project/release/builds?branch=releases/2.9.2 We typically collect it each time we run and the script will capture the latest run.

File: release/release_logs/fetch_release_logs.py (hunk 1 above)
Comment (can-anyscale) [reply release/release_logs/fetch_release_logs.py:41]: Ah yes thanks for fixing these! The new pipeline is perfect.

**Comment by jjyao** [comment]
> REGRESSION 11.32%: 1_n_actor_calls_async (THROUGHPUT) regresses from 8667.010093466013 to 7686.2005077877675 (11.32%) in 2.9.2/microbenchmark.json

I'd rerun this to see if it's real regression: seems it's never below 8k.

**Comment by GeneDer** [comment]
Triggered another job https://buildkite.com/ray-project/release/builds/7612 This is gonna take ~1 day to run, will update the PR once it's completed

**Comment by GeneDer** [comment]
@jjyao @rynewang I have updated the metrics to the latest run. Can you PTAL?

**Review by jjyao — APPROVED** [review]
LG

**Pull request merged**